setup/building-stratum-deps.md

```diff
@@ -0,0 +1,185 @@
+# Building the Stratum Dependencies
+
+Stratum is the component of `infrap4d` that implements the P4Runtime and gNMI
+(OpenConfig) services. It requires that a number of third-party libraries
+be built and installed on the system.
+
+This document explains how to build and install the Stratum dependencies.
+
+> **Note**: For the Intel&reg; IPU E2100, see
+[Building Stratum Dependencies for ES2K](building-stratum-deps-es2k.md).
+
+## 1 Prerequisites
+
+### 1.1 CMake
+
+CMake 3.15 or above must be installed.
+
+Avoid versions 3.24 and 3.25. There is an issue in cmake that causes the
+Protobuf build to fail. This problem was fixed in version 3.26.
+
+### 1.2 OpenSSL
+
+OpenSSL 1.1 must be installed. P4 Control Plane uses OpenSSL instead of
+BoringSSL.
+
+## 2 Source Code
+
+There are two ways to obtain the source code: from the open-source repository
+on GitHub, or as a source package.
+
+### 2.1 GitHub Repository
+
+The Stratum dependencies can be built using a script that is part of the
+IPDK networking-recipe. To clone the repository:
+
+```bash
+git clone --recursive https://github.com/ipdk-io/networking-recipe.git ipdk.recipe
+```
+
+You may omit the `--recursive` option if you are only interested in building
+the dependencies.
+
+You may substitute your own local directory name for `ipdk.recipe`.
+
+The build script for the stratum dependencies is in the `setup` directory.
+
+### 2.2 Source Package
+
+You may receive the source code for the Stratum dependencies in the form of
+an RPM file, a Debian file, or a tarball.
+
+If this is case, use the appropriate command to unpack the source files.
+
+## 3 Install Location
+
+You will need to decide where on your system you would like to install the
+dependencies. This location (the "install prefix") must be specified when you
+configure the build.
+
+It is recommended that you _not_ install the dependency libraries in `/usr` or
```

Comment:
do _not_?

Reply:
I knew this form from use, but I had to look it up to provide an explanation.

_It is recommended that you not ..._ is subjunctive. It is correct. One of the sources I browsed says it is considered dated or formal.

_It is recommended that you do not ..._ is indicative. It is also correct. One of the sources I browsed says that "do" is redundant when used with "recommended".

My personal inclination is use _do_ when using the indicative form ("Google recommends that you do not") and omit it when using the subjunctive form ("It is recommended that you not").

I prefer the subjunctive mood in this case. It's stronger than "we suggest" and weaker than "you should", and it seems more appropriate when it's not accompanied by an explanation.